Convert a keycard account to a regular account

saledjenic opened the pull request:
This PR is created to support a conversion of a keycard account to a regular account.

What's done:
- `DeleteKeypair` endpoint added
- convert a regular to a keycard account process updated in a way that previous implementation was not deleting all keystore files for the keypir being migrated
- `ConvertToRegularAccount` endpoint added
- `TestConvertAccount` tests the entire process of converting a regular account to a keycard account and then converting that keycard account back to a regular account

saledjenic committed:
feat(keycard): `DeleteKeypair` endpoint added

saledjenic committed:
feat(keycard): `ConvertToRegularAccount` endpoint added

It converts a keycard user to a regular user, setting all necessary details
that logged in account looks like it has never been a keycard account.

saledjenic committed:
fix(keycard): convert regular to a keycard account process fixed

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://status.im/docs/))?
- [ ] Have you tested changes with mobile?
- [ ] Have you tested changes with desktop?

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (21)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 3dd05d1e | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3133/1/) | 2023-01-26 07:35:22 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230126-073339-3dd05d-pr3133.zip) |
| :heavy_check_mark: | 3dd05d1e | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3133/1/) | 2023-01-26 07:38:32 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230126-073334-3dd05d-pr3133.aar) |
| :heavy_check_mark: | 3dd05d1e | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3133/1/) | 2023-01-26 07:39:58 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230126-073334-3dd05d-pr3133.zip) |
| :heavy_multiplication_x: | 3dd05d1e | [#1](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3133/1/) | 2023-01-26 07:57:22 | ~23 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3133/1//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 626dad76 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3133/2/) | 2023-01-26 11:59:43 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230126-115819-626dad-pr3133.zip) |
| :heavy_check_mark: | 626dad76 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3133/2/) | 2023-01-26 12:02:17 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230126-115819-626dad-pr3133.aar) |
| :heavy_check_mark: | 626dad76 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3133/2/) | 2023-01-26 12:03:46 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230126-115819-626dad-pr3133.zip) |
| :heavy_check_mark: | 626dad76 | [#2](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3133/2/) | 2023-01-26 12:10:32 | ~12 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3133/2//consoleText) |
| | | | | | | |
| :x: | 8cce6568 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3133/3/) | 2023-01-26 20:04:06 | ~33 sec | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3133/3/consoleText) |
| :x: | 8cce6568 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3133/3/) | 2023-01-26 20:04:11 | ~38 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3133/3/consoleText) |
| :x: | 8cce6568 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3133/3/) | 2023-01-26 20:04:23 | ~51 sec | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3133/3/consoleText) |
| :heavy_multiplication_x: | 8cce6568 | [#3](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3133/3/) | 2023-01-26 20:05:39 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3133/3//consoleText) |
| | | | | | | |
| :heavy_multiplication_x: | 81ebe6f9 | [#4](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3133/4/) | 2023-01-26 20:21:33 | ~1 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3133/4//consoleText) |
| :heavy_check_mark: | 81ebe6f9 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3133/4/) | 2023-01-26 20:21:33 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230126-202012-81ebe6-pr3133.zip) |
| :heavy_check_mark: | 81ebe6f9 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3133/4/) | 2023-01-26 20:23:31 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230126-202012-81ebe6-pr3133.zip) |
| :heavy_check_mark: | 81ebe6f9 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3133/4/) | 2023-01-26 20:24:01 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230126-202011-81ebe6-pr3133.aar) |
| | | | | | | |
| :heavy_check_mark: | 003946ce | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3133/5/) | 2023-01-27 07:20:50 | ~59 sec | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230127-071950-003946-pr3133.zip) |
| :heavy_check_mark: | 003946ce | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3133/5/) | 2023-01-27 07:23:13 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230127-071945-003946-pr3133.zip) |
| :heavy_check_mark: | 003946ce | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3133/5/) | 2023-01-27 07:28:59 | ~9 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230127-071945-003946-pr3133.aar) |
| :heavy_multiplication_x: | 003946ce | [#5](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3133/5/) | 2023-01-27 07:31:58 | ~12 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3133/5//consoleText) |
| :heavy_check_mark: | 003946ce | [#6](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3133/6/) | 2023-01-27 08:11:45 | ~18 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3133/6//consoleText) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 17e3fbfa | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3133/6/) | 2023-01-27 10:00:06 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230127-095846-17e3fb-pr3133.zip) |
| :heavy_check_mark: | 17e3fbfa | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3133/6/) | 2023-01-27 10:02:36 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230127-095846-17e3fb-pr3133.zip) |
| :heavy_check_mark: | 17e3fbfa | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3133/6/) | 2023-01-27 10:03:26 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230127-095846-17e3fb-pr3133.aar) |
| :heavy_multiplication_x: | 17e3fbfa | [#7](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3133/7/) | 2023-01-27 10:10:21 | ~11 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3133/7//consoleText) |
| | | | | | | |
| :heavy_check_mark: | 3feb5c18 | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-3133/7/) | 2023-01-27 10:59:49 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-230127-105827-3feb5c-pr3133.zip) |
| :heavy_check_mark: | 3feb5c18 | [#8](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3133/8/) | 2023-01-27 11:00:47 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-3133/8//consoleText) |
| :heavy_check_mark: | 3feb5c18 | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-3133/7/) | 2023-01-27 11:03:32 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-230127-105827-3feb5c-pr3133.aar) |
| :heavy_check_mark: | 3feb5c18 | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-3133/7/) | 2023-01-27 11:03:40 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-230127-105827-3feb5c-pr3133.zip) |

saledjenic committed:
test: `TestConvertAccount` test update

This tests the entire process of converting a regular account to a keycard
account and then converting that keycard account back to a regular account.

For the need of this test I had to improve `DeleteAccount` function, cause the
previous implementation didn't remove account from the keystore cache, but
only from the keystore.

stefandunca commented on account/accounts.go:
```diff
@@ -606,41 +606,6 @@ func (m *Manager) ReEncryptKeyStoreDir(keyDirPath, oldPass, newPass string) erro
-		return ErrCannotLocateKeyFile{fmt.Sprintf("cannot locate account for address: %s", address.Hex())}
-	}
-	return os.Remove(foundKeyFile)
+func (m *Manager) DeleteAccount(address types.Address, password string) error {
```
Nice simplification

rasom commented:
@saledjenic @richard-ramos @stefandunca 
This PR contains breaking changes which make `develop` unusable for mobile app and it doesn't even contain a version bump.

Not cool.

saledjenic commented:
@rasom sorry for the inconvenience. I was usually adding a version if there are db related changes.

What's broken in the mobile app? How can I help with it now?

siddarthkay commented:
@rasom : is this the breaking change ? https://github.com/status-im/status-go/pull/3133/files#diff-9e4173b89d4a4ff45bea4fb436e514e397edd98ae3693ecf73e608fac7106abaR831

siddarthkay commented:
Oh I see we've already patched it : https://github.com/status-im/status-mobile/blob/01523f3a1de24b97425ff00e6fa89811956139e1/modules/react-native-status/android/src/main/java/im/status/ethereum/module/StatusModule.java#L1308

siddarthkay commented:
just now learnt about https://github.com/status-im/status-go/pull/3145, sorry for the ping pong folks!
